app/controllers/staff/reports_state_controller.rb

```diff
@@ -111,10 +111,17 @@ def update
     report.update!(state: :approved)
     report.create_activity key: "report.approve", owner: current_user
     @report_presenter = ReportPresenter.new(report)
+
+    create_new_report(organisation_id: report.organisation.id, fund_id: report.fund.id)
+
     render "staff/reports_state/approve/complete"
   end
 
   private def report
     @report ||= Report.find(params[:report_id])
   end
+
+  private def create_new_report(organisation_id:, fund_id:)
```

Comment:
The only thing I'd worry about here is how it's enforced that only one "modifiable" report (i.e. not approved, so is either in a state where the DP can edit it, or can be moved to such a state) exists per fund and DP at a time. Rather than implementing this as "create a new `Report` and save it", it might be better as "make sure a `Report` with the right properties and state exists", in case this action ends up being executed more than once somehow.

Reply:
I am going to pair with @lozette on this so I'll move this to draft until we've done that.

Reply:
Spoken to you as well now @jcoglan - great talk! Thanks for explaining it all, I've updated this a bit now to defend against duplicate reports being created – let me know what you think.